Seungsu/app.js

```diff
@@ -49,7 +49,103 @@ app.post("/signup", async (req, res, next) => {
     `,
     [name, email, password, profileImage]
   );
-  res.status(200).json({ message: "userCreated" });
+  res.status(201).json({ message: "userCreated" });
+});
+
+app.post("/post", async (req, res, next) => {
+  const { title, content, imageUrl, userId } = req.body;
+  await myDataSource.query(
+    `INSERT INTO posts(
+        title,
+        content,
+        image_url,
+        user_id
+      ) VALUES (?, ?, ?, ?);
+      `,
+    [title, content, imageUrl, userId]
+  );
+
+  res.status(201).json({ message: "postcreated" });
+});
+
+app.get("/posts", async (req, res, next) => {
+  const data = await myDataSource.query(
+    `SELECT
+    p.user_id as userId,
+    users.profile_image as userProfileImage,
+    p.id as postingId,
+    p.image_url as postingImageUrl,
+    p.content as postingContent
+    FROM posts p
+    INNER JOIN users on p.user_id = users.id`
+  );
+  res.status(200).json({ data: data });
+});
+
+app.get("/posts/users/:userId", async (req, res, next) => {
+  const userId = req.params.userId;
+  const data = await myDataSource.query(
+    `SELECT
+    u.id as userId,
+    u.profile_image as userProfileImage,
+    json_arrayagg(
+      json_object('postingId', p.id, 'postingImage', p.image_url, 'postingContent', p.content
+      )) as postings
+      from users u
+      inner join posts p 
+      on u.id = p.user_id
+      where u.id = ${userId}
+      group by u.id;
+      `
+  );
+  res.status(200).json({ data: data });
+});
+
+app.patch("/modify/posts/:postId", async (req, res, next) => {
```

Comment:
```suggestion
app.patch("/posts/:postId", async (req, res, next) => {
```
제 의도는 아예 modify 자체가 필요없다는 뜻이었습니다~  method에 이미 그 의미가 충분히 담겨있기 때문입니다.

실제로 Github에서 제공하는 API 중 Repository를 생성하는 API의 엔드포인트 주소는 다음과 같습니다.
```Shell
POST {github_url}/orgs/{org_name}/repos
```

Reply:
네 확인했습니다.
method 로 확인가능하기때문에 엔드포인트 주소엔 넣을 필요가없어 modify, delete 삭제 했고, 순서만 바꿔 수정과 삭제를 구분했습니다.